src/app/features/meals/meals.component.ts

```diff
@@ -0,0 +1,34 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { MealService } from 'src/app/core/services/meal.service';
+import { Meal } from 'src/app/core/interfaces/meal';
+import { Subscription } from 'rxjs';
+import { ActivatedRoute } from '@angular/router';
+
+@Component({
+  selector: 'meals',
+  templateUrl: './meals.component.html',
+  styleUrls: ['./meals.component.scss'],
+})
+export class MealsComponent implements OnInit, OnDestroy {
+  meals: Meal[] = [];
+  restaurantId: string;
+  mealsSubscription: Subscription;
+  constructor(
+    private mealService: MealService,
+    private route: ActivatedRoute
+  ) {}
+
+  getMeals(): void {
```

Comment:
Pass restaurantId as a parameter and remove it from the component, it seems like you only need it inside of getMeals:

```
getMeals(restaurantId: string): void {
...
}
ngOnInit(): void {
    const restaurantId = this.route.snapshot.params['id'];
    this.getMeals(restaurantId);
}
```